### PR TITLE
feat: add a cmd to request window size

### PR DESCRIPTION
**Pull request by aymanbagabas**
Fixes: https://github.com/charmbracelet/bubbletea/issues/987#issuecomment-2068452658

### CODE REVIEW

**Review by caarlos0 — APPROVED** [review]

File: commands.go
Comment (meowgorithm) [inline commands.go:197]:
```suggestion
// WindowSize is a command that queries the terminal for its current size. It
// delivers the results to Update via a [WindowSizeMsg]. Keep in mind that
// WindowSizeMsgs will automatically be delivered to Update when the [Program]
// starts and when the window dimensions change so in many cases you will not
// need to explicitly invoke this command.
```

**Comment by atticus-sullivan** [comment]
Because the PR got merged only a few hours ago and this issue is 100% certain linked to this PR, I'm writing no new issue but comment on this PR.

Sorry for waiting until after the merge before testing this. In my test, when I send a `tea.WindowSize()` in the `Update()` function of a model, it causes the program to hang up. My analysis shows that the reason behind this most probably (well tbh, I'm pretty certain on this) is because
you call `p.checkResize()` in the `eventLoop()` which in turn calls `p.Send()` which in the end sends on the `p.msgs` channel. The problem is (by default) this channel is not buffered, thus the send blocks until another goroutine reads from it (but the one which would read from it is the one sending on that channel).

I'm not too familiar with the codebase but searching for other occurrences of `p.Send()`, it seems like this function is called in a new goroutine usually. In this case I'm not sure if the `p.checkResize()` should call `go p.Send()` (would also affect the init stuff which I'm not familiar with) or if we should call `go p.checkResize()` in the `eventLoop()`.

I hope I didn't formulate this too complicated otherwise feel free to ask me what I actually mean. Any thoughts on this? Also, didn't someone else had this issue as well?

**Comment by aymanbagabas** [comment]
@atticus-sullivan Thank you for noticing this one! I've opened https://github.com/charmbracelet/bubbletea/pull/1059 to fix this one and add an example to request window size.

**Comment by atticus-sullivan** [comment]
Alright, thanks for the fast reply + fix. Also having an example for this is quite nice :+1:

**Pull request merged**
